Regenerate API reference asciidoc (#412)
netobserv/network-observability-operator: docs/flowcollector-flows-netobserv-io-v1beta1.adoc

| `ipfix`
| `object`

> | `ipfix` [deprecated (*)] - describes the settings related to the IPFIX-based flow reporter when `spec.agent.type` is set to `IPFIX`.

Reviewer: @jotak I think this option is no longer deprecated? Since we are supporting in the 1.4 release? I noted this in my PR too, but I think this needs updating. Apologies for missing it before.

Reviewer 2: Be carefull, the `agent.type` **IPFIX** is still deprecated since it's ingestion from OVN flows.

However the one we are supporting on 1.4 is **IPFIX** `export.type`:
https://issues.redhat.com/browse/NETOBSERV-962

The unsupported mention should me removed there:
```
IPFIX configuration, such as the IP address and port to send enriched IPFIX flows to. Unsupported (*).
```
https://github.com/netobserv/network-observability-operator/blob/release-1.4/docs/flowcollector-flows-netobserv-io-v1beta1.adoc#specexporters

https://github.com/netobserv/network-observability-operator/blob/release-1.4/api/v1beta1/flowcollector_types.go#L819-L823

| SASL authentication configuration. [Unsupported (*)].

| `tls`
| `object`
| TLS client configuration. When using TLS, verify that the address matches the Kafka port used for TLS, generally 9093.

| `topic`
| `string`
| Kafka topic to use. It must exist, NetObserv will not create it.

|===
== .spec.exporters[].kafka.sasl
Description::
+
--
SASL authentication configuration. [Unsupported (*)].
--

Type::
  `object`

[cols="1,1,1",options="header"]
|===
| Property | Type | Description

| `clientIDReference`
| `object`
| Reference to the secret or config map containing the client ID

| `clientSecretReference`
| `object`
| Reference to the secret or config map containing the client secret

| `type`
| `string`
| Type of SASL authentication to use, or `DISABLED` if SASL is not used

|===
== .spec.exporters[].kafka.sasl.clientIDReference
Description::
+
--
Reference to the secret or config map containing the client ID
--

Type::
  `object`

[cols="1,1,1",options="header"]
|===
| Property | Type | Description

| `file`
| `string`
| File name within the config map or secret

| `name`
| `string`
| Name of the config map or secret containing the file

| `namespace`
| `string`

> | Namespace of the config map or secret containing the file. If omitted, assumes same namespace as where NetObserv is deployed. If the namespace is different, the config map or the secret will be copied so that it can be mounted as required.

Reviewer: ```suggestion
| Namespace of the config map or secret containing the file. If omitted, the default is to the same namespace as where Network Observability Operator is deployed. If the namespace is different, the config map or the secret is copied so that it can be mounted as required.
```

| `type`
| `string`
| Type for the file reference: "configmap" or "secret"

|===
== .spec.exporters[].kafka.sasl.clientSecretReference
Description::
+
--
Reference to the secret or config map containing the client secret
--

Type::
  `object`

[cols="1,1,1",options="header"]
|===
| Property | Type | Description

| `file`
| `string`
| File name within the config map or secret

| `name`
| `string`
| Name of the config map or secret containing the file

| `namespace`
| `string`

> | Namespace of the config map or secret containing the file. If omitted, assumes same namespace as where NetObserv is deployed. If the namespace is different, the config map or the secret will be copied so that it can be mounted as required.

Reviewer: Same as previous comment

| `sasl`
| `object`
| SASL authentication configuration. [Unsupported (*)].

| `tls`
| `object`
| TLS client configuration. When using TLS, verify that the address matches the Kafka port used for TLS, generally 9093.

| `topic`
| `string`
| Kafka topic to use. It must exist, NetObserv will not create it.

|===
== .spec.kafka.sasl
Description::
+
--
SASL authentication configuration. [Unsupported (*)].
--

Type::
  `object`

[cols="1,1,1",options="header"]
|===
| Property | Type | Description

| `clientIDReference`
| `object`
| Reference to the secret or config map containing the client ID

| `clientSecretReference`
| `object`
| Reference to the secret or config map containing the client secret

| `type`
| `string`
| Type of SASL authentication to use, or `DISABLED` if SASL is not used

|===
== .spec.kafka.sasl.clientIDReference
Description::
+
--
Reference to the secret or config map containing the client ID
--

Type::
  `object`

[cols="1,1,1",options="header"]
|===
| Property | Type | Description

| `file`
| `string`
| File name within the config map or secret

| `name`
| `string`
| Name of the config map or secret containing the file

| `namespace`
| `string`

> | Namespace of the config map or secret containing the file. If omitted, assumes same namespace as where NetObserv is deployed. If the namespace is different, the config map or the secret will be copied so that it can be mounted as required.

Reviewer: same namespace comment as previous

| `type`
| `string`
| Type for the file reference: "configmap" or "secret"

|===
== .spec.kafka.sasl.clientSecretReference
Description::
+
--
Reference to the secret or config map containing the client secret
--

Type::
  `object`

[cols="1,1,1",options="header"]
|===
| Property | Type | Description

| `file`
| `string`
| File name within the config map or secret

| `name`
| `string`
| Name of the config map or secret containing the file

| `namespace`
| `string`

> | Namespace of the config map or secret containing the file. If omitted, assumes same namespace as where NetObserv is deployed. If the namespace is different, the config map or the secret will be copied so that it can be mounted as required.

Reviewer: Same namespace comment as previous

| `authToken` describes the way to get a token to authenticate to Loki. +
 - `DISABLED` will not send any token with the request. +
 - `FORWARD` will forward the user token for authorization. +

>  - `HOST` [deprecated (*)] - will use the local pod service account to authenticate to Loki. +

Reviewer: ```suggestion
 - `HOST` [deprecated (*)] - uses the local pod service account to authenticate to Loki. +
```

|===
| Property | Type | Description

| `insecureSkipVerify`
| `boolean`

> | insecureSkipVerify allows skipping client-side verification of the provided certificate If set to true, ProvidedCaFile field will be ignored

Reviewer: ```suggestion
| insecureSkipVerify allows skipping client-side verification of the provided certificate. If set to true, the `ProvidedCaFile` field is ignored.
```

| `provided`
| `object`
| TLS configuration when `type` is set to `PROVIDED`.

| `providedCaFile`
| `object`

> | Reference to the CA file will be ignored

Reviewer: ```suggestion
| Reference to the CA file is ignored.
```

| `string`
| Type for the certificate reference: `configmap` or `secret`

|===
== .spec.processor.metrics.server.tls.providedCaFile
Description::
+
--

> Reference to the CA file will be ignored

Reviewer: ```suggestion
Reference to the CA file is ignored.
```

--

Type::
  `object`

[cols="1,1,1",options="header"]
|===
| Property | Type | Description

| `file`
| `string`
| File name within the config map or secret

| `name`
| `string`
| Name of the config map or secret containing the file

| `namespace`
| `string`

> | Namespace of the config map or secret containing the file. If omitted, assumes same namespace as where NetObserv is deployed. If the namespace is different, the config map or the secret will be copied so that it can be mounted as required.

Reviewer: Same namespace comment as previous.
